extensions/pkg/controller/worker/genericactuator/actuator.go

```diff
@@ -252,6 +254,26 @@ func (a *genericActuator) shallowDeleteMachineClassSecrets(ctx context.Context,
 	return nil
 }
 
+// removeFinalizerFromWorkerSecretRef removes the MCM finalizers from the secret that is referenced by the worker
+func (a *genericActuator) removeFinalizerFromWorkerSecretRef(ctx context.Context, logger logr.Logger, worker *extensionsv1alpha1.Worker) error {
+	logger.Info("Removing MCM finalizers from worker`s secret")
+	secret, err := kutil.GetSecretByReference(ctx, a.client, &worker.Spec.SecretRef)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	if controllerutil.ContainsFinalizer(secret, mcmFinalizer) {
+		controllerutil.RemoveFinalizer(secret, mcmFinalizer)
+	}
+	if controllerutil.ContainsFinalizer(secret, mcmProviderFinalizer) {
+		controllerutil.RemoveFinalizer(secret, mcmProviderFinalizer)
+	}
+
+	return nil
```

Comment:
You need to update/patch the secret back to the API.

Reply:
This is not yet resolved, right?

Reply:
It is, he has switched from `controllerutil` to `controllerutilS`.

Reply:
😕 👀  Is this better? Now we are making two separate calls instead of collecting all changes and doing doing only one call. Maybe this situation cannot occur in reality, but still, seems a little odd to me.

Reply:
My concern was that there was no `patch` request previously with https://github.com/gardener/gardener/commit/51b71c68f19a38f2c96f4caaed33c7450eea95fe.